gum/gumdarwinmodule.c

```diff
@@ -1896,6 +1897,16 @@ const gchar *
 gum_darwin_module_get_dependency_by_ordinal (GumDarwinModule * self,
                                              gint ordinal)
 {
+  switch (ordinal)
+  {
+    case GUM_BIND_SPECIAL_DYLIB_SELF:
+      return self->name;
+    case GUM_BIND_SPECIAL_DYLIB_MAIN_EXECUTABLE:
+    case GUM_BIND_SPECIAL_DYLIB_FLAT_LOOKUP:
+    case GUM_BIND_SPECIAL_DYLIB_WEAK_LOOKUP:
+      return NULL;
+  }
+
   gint i = ordinal - 1;
```

Comment:
This declaration should be moved up, but leaving the assignment here.

Reply:
(Or I suppose we could move the assignment down to right above where it's first used.)